### DISCUSSION TITLE
Should activitylog assume changed for properties it cannot compare?

### DISCUSSION THREAD

**Discussion by danabrey** [spatie/laravel-activitylog#868] [General]
Ran into an issue today related to trying to log changes to a DTO-casted property that includes a `DateInterval`, which cannot be compared by PHP using the spaceship operator and thus causes an `ErrorException`.

In general we use this approach to all of our models for logging changes:

```
    protected static $logAttributes = ['*'];

    protected static $logAttributesToIgnore = ['created_at', 'updated_at', 'deleted_at'];

    protected static $logOnlyDirty = true;
```

Of course, we could add this property to `logAttributesToIgnore` and never log changes to it.

However, I wonder if activitylog should catch the exception and assume that a property that cannot be compared has changed? I'm posting as a discussion as I'm not even sure if I'm right, and this isn't a bug or feature request but more of a question!

**Comment by Gummibeer** [comment]
Hey,

I think that we could add a `CarbonInterval::make($old)->equalTo($new)` check to handle intervals correctly. 🤔
@nagi1 could you add this to v4 #866 ? 🚀

Something like that should handle all cases.
```php
if ($old instanceof DateInterval) {
  return CarbonInterval::make($old)->equalTo($new);
} elseif ($new instanceof DateInterval) {
  return CarbonInterval::make($new)->equalTo($old);
}
```

**Reply by danabrey** [reply]
This sounds great!
